qiskit/pulse/channels/channel_store.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Store of channel registers.
+"""
+import logging
+from typing import List
+
+from qiskit.pulse import ChannelsError
+from .channel_register import ChannelRegister, AcquireChannelRegister, SnapshotChannelRegister
+from .output_channel import DriveChannel, ControlChannel, MeasureChannel
+from .output_channel_register import (DriveChannelRegister,
+                                      ControlChannelRegister,
+                                      MeasureChannelRegister)
+from .pulse_channel import AcquireChannel, SnapshotChannel
+
+logger = logging.getLogger(__name__)
+
+
+class ChannelStore:  # TODO: better name?
+    """Implement a channel store, which is a collection of channel registers.
+    It must be defined before constructing schedule."""
+
+    @classmethod
+    def create_from(cls, backend):
+        """
+        Create channel store with default values in backend configuration.
+        Args:
+            backend(Backend):
+        """
+        config = backend.configuration()
+
+        # system size
+        n_qubit = config.n_qubits
+
+        # frequency information
+        qubit_lo_freqs = config.defaults['qubit_freq_est']
+        qubit_lo_range = config.qubit_lo_range
+        meas_lo_freqs = config.defaults['meas_freq_est']
+        meas_lo_range = config.meas_lo_range
+
+        # generate channel registers
+        channels = [
+            DriveChannelRegister(size=n_qubit, lo_freqs=qubit_lo_freqs),
+            ControlChannelRegister(size=n_qubit),
+            MeasureChannelRegister(size=n_qubit, lo_freqs=meas_lo_freqs),
+            AcquireChannelRegister(size=n_qubit),
+            SnapshotChannelRegister(size=n_qubit)
+        ]
+
+        return ChannelStore(channels)
+
+    def __init__(self, registers: List[ChannelRegister]):
```

Comment:
I think this should have a signature `def __init__(n_qubits, n_composite_channels, drive_channel_los=None, measure_channel_los=None, qubit_lo_range=None, meas_lo_range=None, buffer=0)` and then `from_backend` can just supply these parameters.

Reply:
Good suggestion. I'll think again about the interface of ChannelStore's constructor.

Reply:
I also like the ability to create the registers explicitly. Possibly a constructor like 
```python
def __init__(registers=None, n_qubits=None, n_composite_channels=None, drive_channel_los=None, measure_channel_los=None, qubit_lo_range=None, meas_lo_range=None, buffer=0)
```
would solve this, but it also could be confusing giving this much flexibility.

Reply:
Looking at https://github.com/Qiskit/qiskit-terra/issues/1953, it seems that we are moving towards a class method initializer interface.